File: web/nexus-ui/src/features/auth/register/models/register-user.request.ts
export interface RegisterRequest {
    // Общая информация об аккаунте
    login: string;
    userName: string;
    email: string;
    idGender: string | null;
    idCountry: string | null;
    // Srp
    encryptedVerifier: string;
    srpSalt: string;
    srpVersion: number;
    encryptedVerifierWrapKey: string;
    keyWrapVersion: number; 
    asymmetricKeyId: string;
    // Dek
    encryptedDek: string;
    dekSalt: string;
    cryptoVersion: number;
    // RecoveryKeys
    recoveryKeys: Array<{encryptedValue: string, cryptoVersion: number}>;
}

File: backend/src/shared/contracts/Shared.Contracts/UserManagement/Requests/RegisterUserRequest.cs
namespace Shared.Contracts.UserManagement.Requests
{
    public record RegisterUserRequest(
        // Общая информация об аккаунте
        string Login, 
        string UserName, 
        string Email, 
        string? IdGender, 
        string? IdCountry,
        // Srp
        string EncryptedVerifier, 
        string SrpSalt, 
        int SrpVersion, 
        int SrpCryptoVersion,
        string EncryptedVerifierWrapKey, 
        int KeyWrapVersion, 
        string AsymmetricKeyId,
        // Dek 
        string EncryptedDek, 
        string DekSalt, 
        int CryptoVersion,
        // RecoveryKeys
        IReadOnlyCollection<RecoveryKeyData> RecoveryKeys);

    public record RecoveryKeyData(string EncryptedValue, int CryptoVersion);
}